exec/java-exec/src/main/java/org/apache/drill/exec/record/RecordBatchSizer.java

```diff
@@ -639,6 +641,8 @@ public ColumnSize getColumn(String name) {
   // columns can be obtained from children of topColumns.
   private Map<String, ColumnSize> columnSizes = CaseInsensitiveMap.newHashMap();
 
+  private List<ColumnSize> columnSizesList = new ArrayList<>();
```

Comment:
Why do this? the columnSizes map could return a Collection<ColumnSize> (through java.util.Map.values()) efficiently  (that is, in place).

Reply:
It's for convenience. VectorContainers expose their ValueVectors based on index, so it is handy to be able to lookup column size based on the same index as well. Otherwise I would have to get the Materialized field for the a column, extract the name, and then lookup the ColumnSize in the map.

Reply:
I see; can you please add a comment indicating this behavior (especially within the columnsList() API).

Thanks!

Reply:
done.